openfeature/interfaces.go

```diff
@@ -0,0 +1,50 @@
+package openfeature
+
+import "context"
+
+// IOFApi defines the OpenFeature API contract
+type IOFApi interface {
```

Comment:
This name is not very idiomatic. Can we call it something like `Evaluator` (since it wants to model the Evaluation API?)

Reply:
Agree, I will change the naming

Reply:
I am changing this to `IEvaluation`

Reply:
I guess that's slightly better, the `I` prefix is still not very idiomatic though - that feels very "Java" and not at all like a Go interface name.